pandas/core/base.py

```diff
@@ -1270,6 +1270,13 @@ def searchsorted(
         sorter: NumpySorter = None,
     ) -> npt.NDArray[np.intp] | np.intp:
 
+        if isinstance(value, ABCDataFrame):
+            msg = (
+                "Value must be array-like or scalar, "
```

Comment:
can you add "1D" here

Reply:
I have added 1-D. In fact, n-D object except DataFrame works well with this function. As long as we don't call `pd.array(n-d DataFrame)`.